libs/api/ebpf_api.cpp

```diff
@@ -948,10 +1710,10 @@ ebpf_program_load(
         }
 
         for (auto& program : new_object->programs) {
-            _ebpf_programs.insert(std::pair<fd_t, ebpf_program_t*>(program->fd, program));
+            _ebpf_programs.insert(std::pair<ebpf_handle_t, ebpf_program_t*>(program->handle, program));
         }
         for (auto& map : new_object->maps) {
-            _ebpf_maps.insert(std::pair<fd_t, ebpf_map_t*>(map->map_fd, map));
+            _ebpf_maps.insert(std::pair<ebpf_handle_t, ebpf_map_t*>(map->map_handle, map));
```

Comment:
Created https://github.com/microsoft/ebpf-for-windows/issues/371 to consider populating handle -> map cache at map creation time.